test/TrueLayer.AcceptanceTests/PaymentTests.cs

```diff
@@ -63,6 +63,23 @@ var getPaymentResponse
             payment.Id.ShouldNotBeNullOrWhiteSpace();
             payment.CreatedAt.ShouldNotBe(default);
             payment.PaymentMethod.AsT0.ShouldNotBeNull();
+            if (payment.PaymentMethod.AsT0.ProviderSelection.IsT0)
```

Comment:
Would a Match be more appropriate here than an if/else? IMO it leverages the OneOf structure better

Reply:
the problem is that `Match<TResulkt>()` requires a return value which is not available here unless I return a `bool` value just for the sake of it.

Reply:
Sorry, I meant `Switch`

Reply:
nevermind, found that there is a `Switch` option as well

Reply:
Addressed in [86d5849](https://github.com/TrueLayer/truelayer-dotnet/pull/136/commits/86d5849a14c112ce1520ee1476c6d5d51f90297d)